pkg/plugin/standalone/v1/source.go

```diff
@@ -112,10 +112,6 @@ func (s *sourcePluginClient) Ack(_ context.Context, p record.Position) error {
 }
 
 func (s *sourcePluginClient) Stop(ctx context.Context) (record.Position, error) {
-	if s.stream == nil {
-		return nil, plugin.ErrStreamNotOpen
-	}
```

Comment:
Quick question: how does this help with the blocking call?

Reply:
It doesn't, it's an unrelated thing I noticed. I'm not sure why I added it in the first place, it makes the plugin adapter just a bit too smart, so I removed it. Essentially this `if` prevents `Stop` from being called if the stream is not open. But there's actually nothing stopping us from forwarding the call, because technically we don't need the stream to be open for `Stop` to be called (it's not special, it's a normal function), so the plugin shouldn't prevent us from doing that.

I just noticed it because the test cases were failing for `Stop` because the stream is not opened in those tests.